Modules/_posixsubprocess.c

```diff
@@ -677,6 +705,47 @@ subprocess_fork_exec(PyObject* self, PyObject *args)
         cwd_obj2 = NULL;
     }
 
+    if (groups_list != Py_None) {
+        Py_ssize_t i;
+        int gid;
+
+        if (!PyList_Check(groups_list)) {
+            PyErr_SetString(PyExc_TypeError, "setgroups argument must be a sequence");
```

Comment:
must be a list (as we're being pedantic about a specific type in this internal API)